### DISCUSSION TITLE
Is there any kind of support of .sqlite database with tiles to work offline?

### DISCUSSION THREAD

**Discussion by doubush** [mapbox/mapbox-navigation-ios#3571] [Q&A]
Hello,

I have the android app with maps (uses osmdroid library) and it works with offline maps stored in sqlite database. 
Inside a DB there are OSM tiles.
Can i somehow integrate it with Mapbox? so it works offline and uses sqlite db as a source of tiles?

**Comment by 1ec5** [comment]
Offline maps for Android are managed by the [Mapbox Maps SDK for Android](https://github.com/mapbox/mapbox-maps-android/). This repository is for the Mapbox Navigation SDK for iOS (which uses the Mapbox Maps SDK for iOS).

For what it’s worth, it’s unlikely that the osmdroid database schema matches the (undocumented) schema used by the map SDK.

**Reply by doubush** [reply]
Yes, i want to use Mapbox for iOS. I am creating the clone of my android app for iOS now. And i want on iOS to reuse maps i already used on android .

I thought maybe there is some kind of flexible solution, like "custom" source of tiles, created with a defined interface. So, i could create it and just use sqlite DB to get tiles by x,y,z values ?
